identity-wallet/src/state/credentials/mod.rs

```diff
@@ -54,45 +59,90 @@ impl TryFrom<serde_json::Value> for VerifiableCredentialRecord {
 
     fn try_from(verifiable_credential: serde_json::Value) -> Result<Self, AppError> {
         let display_credential = {
-            let credential_display = get_unverified_jwt_claims(&verifiable_credential)?["vc"].clone();
-
-            // Derive the hash from the credential display.
-            let hash = {
-                let type_key = "type";
-                let type_value = credential_display[type_key].clone();
-
-                let credential_subject_key = "credentialSubject";
-                let mut credential_subject_value = credential_display[credential_subject_key].clone();
-
-                // TODO(ngdil): Remove this hard-coded logic.
-                // Remove the `Passport Number` and `Staff Number` from the credential subject if they exists.
-                credential_subject_value["Passport Number"].take();
-                credential_subject_value["Staff Number"].take();
-                credential_subject_value["achievement"]["id"].take();
-
-                sha256::digest(
-                    json!(
-                        {
-                            "type": type_value,
-                            "credentialSubject": credential_subject_value,
-                        }
+            // Try to parse the Verifiable Credential as an SD-JWT credential.
+            let (id, format, data, issuance_date) = if let Some(sd_jwt_vc) = verifiable_credential
+                .as_str()
+                .and_then(|verifiable_credential| verifiable_credential.parse::<SdJwtVc>().ok())
+            {
+                info!("Verifiable Credential parsed as a SD-JWT VC");
+
+                let issuance_date = sd_jwt_vc.claims().iat.map(|iat| iat.to_rfc3339()).unwrap_or_default();
+                let mut credential_subject = sd_jwt_vc
+                    .clone()
+                    .into_disclosed_object(&Sha256Hasher::new())
+                    .map_err(|_| AppError::Error("Failed to convert SD JWT VC to Disclosed Object".to_string()))?;
+
+                // Remove the SD-JWT specific fields that should not be displayed in the frontend.
+                for key in ["iss", "nbf", "exp", "status", "iat", "sub", "_sd_alg", "cnf", "vct"] {
```

Comment:
Could it lead to unexpected behavior if all of these fields are removed (even for "duplicate detection in demo cases")? What if one of my short-lived credentials gets prolonged by the issuer where only the `nbf`, `exp` and `iat` would be different? Would the "new one" then show up or the old one or actually both? If unsure, please add a TODO and open an issue for tracking.

Reply:
Okay, let's postpone until we actually introduce UUIDs fully.